library/src/main/java/org/xmtp/android/library/PrivatePreferences.kt

```diff
@@ -0,0 +1,142 @@
+package org.xmtp.android.library
+
+import uniffi.xmtpv3.FfiConsent
+import uniffi.xmtpv3.FfiConsentEntityType
+import uniffi.xmtpv3.FfiConsentState
+
+enum class ConsentState {
+    ALLOWED,
+    DENIED,
+    UNKNOWN;
+
+    companion object {
+        fun toFfiConsentState(option: ConsentState): FfiConsentState {
+            return when (option) {
+                ALLOWED -> FfiConsentState.ALLOWED
+                DENIED -> FfiConsentState.DENIED
+                else -> FfiConsentState.UNKNOWN
```

Comment:
I ran into this in Rust too when matching between the proto and native consent state and was tempted to do `_ -> UNKNOWN`, but it's probably better to be explicit and do `UNKNOWN -> UNKNOWN` so this isn't a footgun in the future when we might add another consent state and the compiler happily matches it to `UNKNOWN`.